merino-cache/src/memory.rs

```diff
@@ -174,6 +176,13 @@ impl Suggester {
             ?removed_storage,
             "finished removing expired entries from cache"
         );
+
+        metrics_client
+            .count("cache.memory.storage_len", items.len_storage() as i64)
```

Comment:
I missed this the first time, but this needs to be `metrics_client.gauge`, not `.count`. That's because `count` is equivalent to `incr`, but with a larger step. (more specifically, `incr` is `count` with n = 1). With count we'd end up with a graph of the integral of the cache size overtime, which definitely isn't what we want.